### DISCUSSION TITLE
Cannot get jdeploy to work for me

### DISCUSSION THREAD

**Discussion by JimT787** [shannah/jdeploy#91] [Q&A]
Spent the past few days trying to get a simple app to publish using to jdeploy to no avail.

Is anybody else having this problem...?

I installed NPM and Jdeploy in accordance with the "Getting Started" page.  That all seems to go good.  I made the JAR executable so I can just use java -jar <xxx> to run it from the command line.

I used the GUI to produce the package.json file and made a repository on npm called "jimjdeploytest"  

It published ok and said I could go to "https://www.jdeploy.com/~jimjdeploytest" and that website looks ok.  When I download the .exe installer for windows, unzip it and click the installer, I get a quick screen flash and that's it.  Yesterday, it seemed it was working better and would install, but the app itself would not run.  I could navigate to the .jdeploy folder and again, run the app itself using "java -jar <app.jar>" but the jdeploy icon on the desktop will not do that it--it just does a quick screen flash and then nothing.  

I also have Java 17 specified in the package.json, but all it seems to want to install (if it install anything in the .jdeploy folder) is Java 11.

The github repository for this app is:  "https://github.com/JimT787/jim-jdeploy-test-npm"

It's basically just a simple Hello World app using JavaFX as a test before I use JDeploy to publish a different app.  I have been using JPackage which has been working fine, but need 3 computers for all the end users (Windows, Mac/Intel, Mac/Apple Silicon).

JPackage looks awsome and would love to use it...  any help?

Thanks a bunch!

**Comment by shannah** [comment]
Thanks for posting this.  The problem is that the javaVersion property needs to be a "string" in the package.json.
E.g. `"javaVersion": "17"` not `"javaVersion": 17`.

Did you specify this version with the jDeploy GUI or was it from manually editing the package.json?

**Reply by JimT787** [reply]
I used "jdeploy init" from the command line.  I just tried it again and it sets up an integer for the jdk rather than a string.  Here is the screenshot:
![integer-jdk-highlighted](https://user-images.githubusercontent.com/98830368/218176924-6c509c69-b1a4-4f1a-af27-d3fe19341179.png)

**Reply by JimT787** [reply]
Oops--I think that from version 4.0.1 and you just put out 4.0.2, so probably fixed...sorry about that...

**Comment by shannah** [comment]
Actually, thanks for reporting this.  I did find a few places it is setting the javaVersion to an int.  I have corrected that and made a new release.
I still need to investigate why this wasn't picked up before...
